tensorboard/webapp/widgets/line_chart_v2/line_chart_component.ts

```diff
@@ -0,0 +1,348 @@
+  ElementRef,
+  Input,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+
+import {MainThreadChart} from './lib/chart';
+import {Chart} from './lib/chart_types';
+import {
+  ChartCallbacks,
+  ChartOptions,
+  DataSeries,
+  DataSeriesMetadataMap,
+  Extent,
+  RendererType,
+  Scale,
+  ScaleType,
+} from './lib/public_types';
+import {createScale} from './lib/scale';
+import {
+  areExtentsEqual,
+  isOffscreenCanvasSupported,
+  isWebGl2Supported,
+} from './lib/utils';
+import {WorkerChart} from './lib/worker/worker_chart';
+import {
+  computeDataSeriesExtent,
+  getRendererType,
+} from './line_chart_internal_utils';
+import {TooltipTemplate} from './sub_view/line_chart_interactive_view';
+
+export {TooltipTemplate} from './sub_view/line_chart_interactive_view';
+
+const DEFAULT_EXTENT: Extent = {x: [0, 1], y: [0, 1]};
+
+interface DomDimensions {
+  main: {width: number; height: number};
+  yAxis: {width: number; height: number};
+  xAxis: {width: number; height: number};
+}
+
+@Component({
+  selector: 'line-chart',
+  templateUrl: 'line_chart_component.ng.html',
+  styleUrls: ['line_chart_component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LineChartComponent implements AfterViewInit, OnChanges, OnDestroy {
+  readonly RendererType = RendererType;
+
+  @ViewChild('main', {static: true, read: ElementRef})
+  private main!: ElementRef<HTMLElement>;
+
+  @ViewChild('xAxis', {static: true, read: ElementRef})
+  private xAxis!: ElementRef<HTMLElement>;
+
+  @ViewChild('yAxis', {static: true, read: ElementRef})
+  private yAxis!: ElementRef<HTMLElement>;
+
+  @ViewChild('chartEl', {static: false, read: ElementRef})
+  private chartEl?: ElementRef<HTMLCanvasElement | SVGElement>;
+
+  @Input()
+  preferredRendererType: RendererType = isWebGl2Supported()
+    ? RendererType.WEBGL
+    : RendererType.SVG;
+
+  @Input()
+  seriesData!: DataSeries[];
+
+  // In case of PR curve line chart, we do not want to compute the viewBox based on the
+  // data.
+  @Input()
+  fixedViewBox?: Extent;
+
+  @Input()
+  seriesMetadataMap!: DataSeriesMetadataMap;
+
+  @Input()
+  xScaleType: ScaleType = ScaleType.LINEAR;
+
+  @Input()
+  yScaleType: ScaleType = ScaleType.LINEAR;
+
+  @Input()
+  tooltipTemplate?: TooltipTemplate;
+
+  xScale: Scale = createScale(this.xScaleType);
+  yScale: Scale = createScale(this.xScaleType);
+  viewBox: Extent = DEFAULT_EXTENT;
+
+  domDimensions: DomDimensions = {
+    main: {width: 0, height: 0},
+    xAxis: {width: 0, height: 0},
+    yAxis: {width: 0, height: 0},
+  };
+
+  private lineChart?: Chart;
+  private dataExtent: Extent = DEFAULT_EXTENT;
+  private isDataUpdated = false;
+  private isMetadataUpdated = false;
+  // Must set the default view extent since it is an optional input.
+  private isViewBoxUpdated = true;
+  private isFixedViewBoxUpdataed = false;
+  private maybeSetViewBoxToDefault = true;
+  private scaleUpdated = false;
```

Comment:
Do we need to set this to true, since these are optional `@Input`s to LineChartComponent, similar to L124?